libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -822,6 +825,23 @@ string YulUtilFunctions::arrayAllocationSizeFunction(ArrayType const& _type)
 	});
 }
 
+string YulUtilFunctions::structAllocationSizeFunction(StructType const& _type)
+{
+	solAssert(_type.dataStoredIn(DataLocation::Memory), "");
+	string functionName = "struct_allocation_size_" + _type.identifier();
+	return m_functionCollector.createFunction(functionName, [&]() {
+		Whiskers w(R"(
+			function <functionName>(memoryMemberCount) -> size {
+				// Make sure we can allocate memory without overflow
+				if gt(memoryMemberCount, 0xffffffffffffffff) { revert(0, 0) }
```

Comment:
I know we use revert a lot in such situations, but that should actually be `invalid()` - but I guess this should be handled as part of a different issue.

Reply:
Created https://github.com/ethereum/solidity/issues/8556 to track that